src/plugins/formulas/utils.js

```diff
@@ -1,63 +1,30 @@
-/**
- * Check if provided expression is valid formula expression.
- *
- * @param {*} expression Expression to check.
- * @returns {Boolean}
- */
-export function isFormulaExpression(expression) {
-  return typeof expression === 'string' && expression.length >= 2 && expression.charAt(0) === '=';
-}
+// TODO: docs
+export function parseHFValue(value) {
+  if (value && value.constructor && value.constructor.name.includes('Error')) {
+    return parseErrorObject(value);
 
-/**
- * Check if provided formula expression is escaped.
- *
- * @param {*} expression Expression to check.
- * @returns {Boolean}
- */
-export function isFormulaExpressionEscaped(expression) {
-  return typeof expression === 'string' && expression.charAt(0) === '\'' && expression.charAt(1) === '=';
-}
+  } else if (typeof value === 'symbol') { // TODO: parsing `Symbol`s received from HF to `null`, needs to be checked if `Symbol` always means `null`.
+    return null;
+  }
 
-/**
- * Replace escaped formula expression into valid string.
- *
- * @param {String} expression Expression to process.
- * @returns {String}
- */
-export function unescapeFormulaExpression(expression) {
-  return isFormulaExpressionEscaped(expression) ? expression.substr(1) : expression;
+  return value.toString(); // TODO: currently parsing everything to strings
 }
 
-/**
- * Upper case formula expression.
- *
- * @param {String} expression Formula expression.
- * @returns {String}
- */
-export function toUpperCaseFormula(expression) {
-  const PATTERN = /(\\"|"(?:\\"|[^"])*"|(\+))|(\\'|'(?:\\'|[^'])*'|(\+))/g;
-  const strings = expression.match(PATTERN) || [];
-  let index = -1;
+// TODO: docs
+export function parseErrorObject(errorObject) {
+  let parsedErrorValue = '';
 
-  return expression.toUpperCase().replace(PATTERN, () => {
-    index += 1;
-
-    return strings[index];
-  });
-}
+  if (errorObject.type === 'DIV_BY_ZERO') {
```

Comment:
Don't you think that this should be a part of `CellError` error type? Maybe as a `message` or `value` property next to the `type` ?

https://github.com/handsontable/hyperformula/blob/e696d1d8226ca80d2f5c653771d52dd4857a573d/src/Cell.ts#L79-L82

Moreover it should be localized with:
https://github.com/handsontable/hyperformula/blob/e696d1d8226ca80d2f5c653771d52dd4857a573d/src/i18n/plPL.ts#L42-L50

Reply:
I think that's a good idea. 👍

Reply:
Reported to HyperFormula repo: https://github.com/handsontable/hyperformula/issues/48